go/vt/tabletserver/tabletserver.go

```diff
@@ -967,89 +971,79 @@ func (tsv *TabletServer) SplitQueryV2(
 	numRowsPerQueryPart int64,
 	algorithm querypb.SplitQueryRequest_Algorithm,
 ) (splits []querytypes.QuerySplit, err error) {
-	logStats := newLogStats("SplitQuery", ctx)
-	logStats.OriginalSQL = sql
-	logStats.BindVariables = bindVariables
-	defer tsv.handleError(sql, bindVariables, &err, logStats)
-	if err = tsv.startRequest(target, false, false); err != nil {
-		return nil, err
-	}
-	// We should not set a timeout for SplitQueryV2. So, we can't use execRequest.
-
-	// SplitQuery using the Full Scan algorithm can take a while and
-	// we don't expect too many of these queries to run concurrently.
-	defer tsv.endRequest(false)
-
-	ciSplitColumns := make([]sqlparser.ColIdent, 0, len(splitColumns))
-	for _, s := range splitColumns {
-		ciSplitColumns = append(ciSplitColumns, sqlparser.NewColIdent(s))
-	}
+	err = tsv.execRequest(
+		ctx, 24*365*10*time.Hour, /* practically forever */
```

Comment:
Nit: Can we use a nil value to signify no timeout instead? It'll be a little cleaner.